sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/PublicMethodArgumentsShouldBeCheckedForNull.cs

```diff
@@ -64,5 +64,26 @@ public void CompliantCases(bool b, object o1, object o2, object o3, object o4, E
 
             Program.staticField.ToString(); // Compliant
         }
+
+        public void MoreCompliantCases(string s1, string s2)
+        {
+            if (string.IsNullOrEmpty(s1))
```

Comment:
Are there any tests for e.g. _!string.IsNullOrEmpty(s1)_ ?

Reply:
It should not be required, as the `!` is not part of the invocation. The negation is handled in a separate node of the exploded graph.

Reply:
I think it is still useful to add a test to be sure this is covered. If not it's probably not needed to cover it right now.